### PR TITLE
Add io.github.JakubMelka.Pdf4qt

### PR DESCRIPTION
**Pull request by jointri**
I want to apologize, I made a very rash and impulsive decision to close and delete my first PR. This is the new one which I'm certain will be my last one for this app.

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by jointri** [comment]
bot, build io.github.JakubMelka.Pdf4qt

File: io.github.JakubMelka.Pdf4qt.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    sources:
+      - type: archive
+      - url: https://github.com/uclouvain/openjpeg/archive/refs/tags/v2.5.0.tar.gz
+      - sha256: 0333806d6adecc6f7a91243b2b839ff4d2053823634d4f6ed7a59bc87409122a
+  - name: lcms
+    sources:
+      - type: archive
+        url: https://github.com/mm2/Little-CMS/archive/refs/tags/lcms2.15.tar.gz
+        sha256: 30d928c9b564c670b6b57c917e5c2599247b713ba6cf2dbd089a8efa70a1952f
+    - name: pdf4qt
```
Comment (hfiguiere) [inline io.github.JakubMelka.Pdf4qt.yml:23]: this indentation is incorrect.

you chose to use yaml, you must make sure your indentations are consistent.

File: io.github.JakubMelka.Pdf4qt.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  - --filesystem=home
+  - --share=network
+modules:
+  - name: openjpeg
+    buildsystem: cmake-ninja
+    sources:
+      - type: archive
+      - url: https://github.com/uclouvain/openjpeg/archive/refs/tags/v2.5.0.tar.gz
+      - sha256: 0333806d6adecc6f7a91243b2b839ff4d2053823634d4f6ed7a59bc87409122a
+  - name: lcms
```
Comment (hfiguiere) [inline io.github.JakubMelka.Pdf4qt.yml:18]: lcms is in the runtime.

File: io.github.JakubMelka.Pdf4qt.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+app-id: io.github.JakubMelka.Pdf4qt
+runtime: org.kde.Runtime
+runtime-version: 5.15-22.08
+sdk: org.freedesktop.Sdk
```
Comment (hfiguiere) [inline io.github.JakubMelka.Pdf4qt.yml:4]: the SDK doesn't match the runtime.

File: io.github.JakubMelka.Pdf4qt.yml (hunk 1 above)
Comment (jointri) [reply io.github.JakubMelka.Pdf4qt.yml:23]: I'll fix that right now. I'll double check by running it through a yaml linter.

File: io.github.JakubMelka.Pdf4qt.yml (hunk 3 above)
Comment (jointri) [reply io.github.JakubMelka.Pdf4qt.yml:4]: I didn't know if I could use both the kde runtime and the freedesktop runtime together.

File: io.github.JakubMelka.Pdf4qt.desktop
[the remark this answers is not in the document]
Comment (jointri) [reply io.github.JakubMelka.Pdf4qt.desktop:1]: Until upstream includes a desktop file, can I include it myself?

**Comment by jointri** [comment]
bot, build io.github.JakubMelka.Pdf4qt

File: io.github.JakubMelka.Pdf4qt.yml (hunk 3 above)
Comment (hfiguiere) [reply io.github.JakubMelka.Pdf4qt.yml:4]: the KDE runtime "includes" the freedestkop runtime. It's a superset.

File: io.github.JakubMelka.Pdf4qt.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+app-id: io.github.JakubMelka.Pdf4qt
+runtime: org.kde.Runtime
+runtime-version: 5.15.10
+sdk: org.freedesktop.Sdk
+command: pdf4qt
+finish-args:
+  - --socket=fallback-x11
```
Comment (hfiguiere) [inline io.github.JakubMelka.Pdf4qt.yml:7]: you need `share=ipc` for X11 (even fallback)

you also likely need `device=dri` if you do anything with GL.

File: io.github.JakubMelka.Pdf4qt.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  - --socket=wayland
+  - --filesystem=home
+  - --share=network
```
Comment (hfiguiere) [inline io.github.JakubMelka.Pdf4qt.yml:10]: does it really need network access?

File: io.github.JakubMelka.Pdf4qt.yml [hunk 6]
```diff
@@ -0,0 +1,23 @@
+app-id: io.github.JakubMelka.Pdf4qt
+runtime: org.kde.Runtime
+runtime-version: 5.15.10
+sdk: org.freedesktop.Sdk
+command: pdf4qt
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --filesystem=home
```
Comment (hfiguiere) [inline io.github.JakubMelka.Pdf4qt.yml:9]: qt use portals by default. Do you really need full home access?

**Review by hfiguiere — CHANGES_REQUESTED** [review]
I suspect you are missing the AppStream file too.

File: io.github.JakubMelka.Pdf4qt.desktop
Comment (hfiguiere) [inline io.github.JakubMelka.Pdf4qt.desktop:1]: this file is never installed.

File: io.github.JakubMelka.Pdf4qt.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
+app-id: io.github.JakubMelka.Pdf4qt
+runtime: org.kde.Runtime
+runtime-version: 5.15.10
```
Comment (hfiguiere) [inline io.github.JakubMelka.Pdf4qt.yml:3]: this version doesn't exist.

File: io.github.JakubMelka.Pdf4qt.desktop
Comment (jointri) [reply io.github.JakubMelka.Pdf4qt.desktop:1]: I'll tell upstream to include a desktop file.

File: io.github.JakubMelka.Pdf4qt.yml (hunk 5 above)
Comment (hfiguiere) [reply io.github.JakubMelka.Pdf4qt.yml:10]: openssl can be used just for the crypto code. but again, it's a question.

File: io.github.JakubMelka.Pdf4qt.yml [hunk 8]
```diff
@@ -1,23 +1,23 @@
 app-id: io.github.JakubMelka.Pdf4qt
 runtime: org.kde.Runtime
-runtime-version: 5.15.10
+runtime-version: 5.15.22.08
```
Comment (hfiguiere) [inline io.github.JakubMelka.Pdf4qt.yml:3]: this version doesn't exist. It was right initially.

**Comment by jointri** [comment]
What exactly does the upstream need to include in the .desktop file? I want to make sure upstream has it right.

**Comment by hfiguiere** [comment]
what any desktop file should have. It's what make the app appear in the "start menu" or whatever paradigm you DE uses. It's a 20 year old standard, so it's not like it's new. And not specific to flatpak at all. We just require it.

**Comment by jointri** [comment]
Does it matter -where- in the upstream source the .desktop file is located? Or that it's there, -period-?

**Comment by hfiguiere** [comment]
it's required so if upstream doesn't have one it has to be provided. but then upstream should have one for a future release as it is a bug on their side to not have one.

File: io.github.JakubMelka.Pdf4qt.yml (hunk 8 above)
Comment (hfiguiere) [reply io.github.JakubMelka.Pdf4qt.yml:3]: btw this would still be wrong given:

> Download Qt 6.4

File: io.github.JakubMelka.Pdf4qt.yml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+sdk: org.freedesktop.Sdk
+command: pdf4qt
+finish-args:
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=ipc
+  - --device=dri
+modules:
+  - name: openjpeg
```
Comment (hfiguiere) [inline io.github.JakubMelka.Pdf4qt.yml:12]: this is also in the runtime.

File: io.github.JakubMelka.Pdf4qt.yml (hunk 8 above)
Comment (hfiguiere) [reply io.github.JakubMelka.Pdf4qt.yml:3]:
```suggestion
runtime-version: 6.4
```

**Comment by JakubMelka** [comment]
Hello @hfiguiere, I am author of PDF4QT. I would like to provide these Desktop files. I have 4 GUI applications and 1 command line application. Should I provide desktop file for each of this application?

**Comment by hfiguiere** [comment]
one desktop file per app. There should be one considered as the main one. For flatpak (and that's compatible with anything else) they should have a name based on the app-id with the main app being `$FLATPAK_ID.desktop`. That main app is also the one specified in `command` for the manifest. The cli app doesn't need one.

And you build system should install them in the standard location (honouring prefix, off course).

**Comment by hfiguiere** [comment]
and they should all have a different icon. It's not mandatory per-see, but it's good for the user experience.

File: io.github.JakubMelka.Pdf4qt.yml (hunk 8 above)
Comment (hfiguiere) [reply io.github.JakubMelka.Pdf4qt.yml:3]: `6.5` is out now.

**Comment by hfiguiere** [comment]
svg is good.

**Comment by barthalion** [comment]
bot, build io.github.JakubMelka.Pdf4qt

File: io.github.JakubMelka.Pdf4qt.yml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+    buildsystem: cmake-ninja
+    sources:
+        type: archive
```
Comment (hfiguiere) [inline io.github.JakubMelka.Pdf4qt.yml:15]:
```suggestion
      - type: archive
```

`sources` is an array of objects.

File: io.github.JakubMelka.Pdf4qt.yml [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+        url: https://github.com/uclouvain/openjpeg/archive/refs/tags/v2.5.0.tar.gz
+        sha256: 0333806d6adecc6f7a91243b2b839ff4d2053823634d4f6ed7a59bc87409122a
+  - name: pdf4qt
+    buildsystem: cmake-ninja
+    sources:
+        type: archive
```
Comment (hfiguiere) [inline io.github.JakubMelka.Pdf4qt.yml:21]:
```suggestion
      - type: archive
```

same

**Comment by JakubMelka** [comment]
Hello @hfiguiere , I would like to have this done. Can I help with something?

**Comment by hfiguiere** [comment]
nothing I can do here. 

It's all in the hands of the submitter.

Or you can submit it yourself (new PR) that's we can review and move forward.

**Pull request closed without merging**